_posts/2025-08-15-spec-driven-development-in-action.adoc

```diff
@@ -0,0 +1,106 @@
+
+=== Step 2: Launch the Workspace with a Devfile
+
+The `devfile.yaml` is the key to a one-click setup. It tells Eclipse Che to create a workspace using a pre-configured container image (`quay.io/che-incubator/cli-ai-tools:latest`) that already includes **RA.Aid** and **Aider**. This means your development environment comes ready with all the specialized AI tools installed.
+
+=== Step 3: Secure Your AI's Access
+
+To work, the AI assistants need access to a Large Language Model (LLM). **RA.Aid** and **Aider** are flexible and support many different providers, including Gemini, OpenAI, Anthropic, Groq, and more. For this demo, we'll use the Gemini API.
+
+The recommended way to provide credentials is by creating a Kubernetes `Secret`. Eclipse Che will then automatically mount this secret as an environment variable inside your workspace. For the Gemini API, the required environment variable is `GEMINI_API_KEY`.
+
+You can create the secret for Gemini with the following command:
+
+```sh
+oc apply -f - <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: gemini-api-key
+  labels:
+    controller.devfile.io/mount-to-devworkspace: 'true'
+    controller.devfile.io/watch-secret: 'true'
+  annotations:
+    controller.devfile.io/mount-as: env
+data:
+  GEMINI_API_KEY: [base64 encoded Gemini API key]
+type: Opaque
+EOF
+```
+
+If you prefer to use a different provider, simply consult the **RA.Aid** or **Aider** documentation for the correct environment variable name (e.g., `OPENAI_API_KEY`) and adapt the secret accordingly.
+
+=== Step 4: Unleash the AI
+
+Once you have crafted the `spec.md` for the application of your dreams, all it takes is this one command in the terminal to set the AI to work:
+
+```sh
+ra-aid --use-aider --cowboy-mode --show-cost --track-cost --msg-file spec.md
+```
+
+This command tells **RA.Aid** to use your `spec.md` file as its guide, leverage **Aider** for code modifications, and run autonomously without needing interactive approval (`--cowboy-mode`).
+
+Instantly, the agent will parse the specification and begin building the application right inside your Eclipse Che workspace, just as it did for my URL shortener. See the whole process unfold in the video below.
+
+++++
+<iframe width="560" height="315" src="https://www.youtube.com/embed/L_a1ig1VrxY?si=DGC5q405MhLMnmrY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+++++
+
+As the agent works, you can follow its progress directly in the terminal. The output shows its step-by-step thinking process as it interprets the specification, plans its actions, and executes them.
+
+image::/assets/img/spec-driven-development-in-action/1-ra-aid-command-running.png[RA.Aid command running in the terminal, showing the AI planning and executing steps for the URL shortener.]
```

Comment:
You might want to add an image caption with a title (like `.Figure: ...`) so it’s clearer in the final output.